plugin.py

```diff
@@ -144,6 +146,51 @@ def log_solar_charger(
         print(json.dumps(delta))
         sys.stdout.flush()
 
+    def log_dcdc_converter(
+        self,
+        bl_device: BLEDevice,
+        cfg_device: ConfiguredDevice,
+        data: DcDcConverter,
+        id_: str,
+    ):
+        delta = {
+            "updates": [
+                {
+                    "source": {
+                        "label": "Victron",
+                        "type": "Bluetooth",
+                        "src": bl_device.address,
+                    },
+                    "timestamp": datetime.datetime.utcnow().isoformat() + "Z",
+                    "values": [
+                        {
+                            "path": f"electrical.converters.{id_}.chargingMode",
+                            "value": data.get_charge_state().name.lower(),
+                        },
+                        {
+                            "path": f"electrical.converters.{id_}.chargerError",
+                            "value": data.get_charger_error().name.lower(),
+                        },
+                        {
+                            "path": f"electrical.converters.{id_}.input.voltage",
+                            "value": data.get_input_voltage(),
+                        },
+                        {
+                            "path": f"electrical.converters.{id_}.output.voltage",
+                            "value": data.get_output_voltage(),
+                        },
+                        {
+                            "path": f"electrical.converters.{id_}.chargingMode.offReason",
```

Comment:
Does having a sub-attribute like this work? I would have assumed that chargingMode would need to be an object for it to work.

Also, it won't always be set.

Reply:
technically, you can chain as many path elements together as you want (okay, maybe there's a limit *somewhere*, so make that "as many as you want, within reason"), and not all of them need to have actual content. As @tkurki said in his comment on issue #4 : "the SK server is schema agnostic, you can basically invent your own paths." ...
 
also, get_off_reason() will always return *something*, e.g. if the charger is active, then chargingMode will be either Bulk, Absorption or Float, and the offReason will be "NONE" (see https://github.com/keshavdv/victron-ble/blob/main/victron_ble/devices/base.py#L168-L178 for the OffReason enum)

Reply:
What I mean is: Can you have a value for `chargingMode` and `chargingMode.offReason`? I don't think I've even seen nested values like that.

Reply:
yes, you can. I have the exact code that's in this PR running on my SignalK server, without issues.

this is what the raw SK delta looks like:
```
[
  {
    "updates": [
      {
        "source": {
          "label": "Victron",
          "type": "Bluetooth",
          "src": "REDACTED"
        },
        "timestamp": "2023-04-15T19:27:16.737560Z",
        "values": [
          {
            "path": "electrical.converters.dc12-dc24.chargingMode",
            "value": "off"
          },
          {
            "path": "electrical.converters.dc12-dc24.chargerError",
            "value": "no_error"
          },
          {
            "path": "electrical.converters.dc12-dc24.input.voltage",
            "value": 13.35
          },
          {
            "path": "electrical.converters.dc12-dc24.output.voltage",
            "value": 0
          },
          {
            "path": "electrical.converters.dc12-dc24.chargingMode.offReason",
            "value": "switched_off_register"
          }
        ]
      }
    ],
    "context": "vessels.self"
  }
]
```

That said, I'd have no problem with changing the path to `electrical.converters.{id_}.chargerOffReason` if that's what you're worried about.

Reply:
> That said, I'd have no problem with changing the path to electrical.converters.{id_}.chargerOffReason if that's what you're worried about.

Yeah, I'd prefer that.

I suspect the nested structure would cause trouble in the future, even if it works now.

Reply:
done: see [e12a1ff](https://github.com/stefanor/signalk-victron-ble/pull/1/commits/e12a1ff266bd51af576d10b5e80b5985d28f104d)